tracing/init.go

```diff
@@ -33,17 +31,16 @@ func InitTracer(serviceName, instanceName string) (trace.Tracer, func(), error)
 			log.Fatal(err)
 		}
 	}
-
-	return tp.Tracer(serviceName), shutdown, err
+	return tp, shutdown, err
 }
 
 // tracerProvider returns an OpenTelemetry TracerProvider configured to use
-// the Jaeger exporter that will send spans to the provided url. The returned
+// the OTEL exporter that will send spans to the provided url. The returned
 // TracerProvider will also use a Resource configured with all the information
 // about the application.
 func newTracerProvider(serviceName, instanceName string) (*tracesdk.TracerProvider, error) {
-	// Create the Jaeger exporter
-	exp, err := jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(JaegerCollectorURL)))
+	// Create the OTEL exporter
+	exp, err := otel.New(context.Background(), otel.WithEndpointURL(os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")))
```

Comment:
you are not supposed to read env vars yourself, the OTEL SDK does that automatically.

example https://github.com/jaegertracing/jaeger/blob/8a338003d460ce4dabe425b631f96d08e6ee01b2/examples/hotrod/pkg/tracing/init.go#L90

Reply:
I found an instance in hotrod which does it this way https://github.com/jaegertracing/jaeger/blob/8a338003d460ce4dabe425b631f96d08e6ee01b2/examples/hotrod/pkg/tracing/init.go#L86.

Reply:
Now the OTEL SDK automatically reads the env variable. I have tested this by changing the export variable in the terminal.